src/Features/Core/Portable/AddConstructorParametersFromMembers/State.cs

```diff
@@ -38,29 +30,49 @@ private class State
-                TextSpan textSpan,
-                ImmutableArray<ISymbol> selectedMembers,
-                CancellationToken cancellationToken)
+                ImmutableArray<ISymbol> selectedMembers)
             {
                 if (!selectedMembers.All(IsWritableInstanceFieldOrProperty))
                 {
                     return false;
                 }
 
-                this.TextSpan = textSpan;
-                this.SelectedMembers = selectedMembers;
-                this.ContainingType = this.SelectedMembers[0].ContainingType;
-                if (this.ContainingType == null || this.ContainingType.TypeKind == TypeKind.Interface)
+                ContainingType = selectedMembers[0].ContainingType;
+                if (ContainingType == null || ContainingType.TypeKind == TypeKind.Interface)
                 {
                     return false;
                 }
 
-                this.Parameters = service.DetermineParameters(selectedMembers);
+                var parameters = service.DetermineParameters(selectedMembers);
+                DelegatedConstructor = service.GetDelegatedConstructor(ContainingType, parameters);
 
-                this.MatchingConstructor = service.GetMatchingConstructor(this.ContainingType, this.Parameters);
-                this.DelegatedConstructor = service.GetDelegatedConstructor(this.ContainingType, this.Parameters);
-                return this.DelegatedConstructor != null;
+                if (DelegatedConstructor != null)
+                {
+                    var zippedParametersAndSelectedMember = parameters.Zip(selectedMembers, (parameter, selectedMember) => (parameter, selectedMember));
+                    var missingParamtersBuilder = ImmutableArray.CreateBuilder<IParameterSymbol>();
+                    var missingMembersBuilder = ImmutableArray.CreateBuilder<ISymbol>();
+                    foreach ((var parameter, var selectedMember) in zippedParametersAndSelectedMember)
+                    {
+                        if (IsParameterMissingFromConstructor(DelegatedConstructor, parameter))
+                        {
+                            missingParamtersBuilder.Add(parameter);
+                            missingMembersBuilder.Add(selectedMember);
+                        }
+                    }
+
+                    MissingParameters = missingParamtersBuilder.ToImmutableArray();
+                    MissingMembers = missingMembersBuilder.ToImmutableArray();
+                }
+
+                return DelegatedConstructor != null;
+            }
+
+            /// <summary>
+            /// Find whether <paramref name="parameter"/> is contained in <paramref name="constructor"/>'s parameter list by comparing name.
+            /// </summary>
+            private bool IsParameterMissingFromConstructor(IMethodSymbol constructor, IParameterSymbol parameter)
+            {
+                return !constructor.Parameters.Select(p => p.Name).Contains(parameter.Name);
```

Comment:
you are creating the name list every time this is called.  instead, pull this into the outer loop, create the name list once, then check it for the parameter name for each parameter you care about.

Reply:
this still applies.

Reply:
I just realize I omit this one :)